src/internal/connector/onedrive/collections.go

```diff
@@ -404,13 +405,45 @@ func (c *Collections) UpdateCollections(
 
 		switch {
 		case item.GetFolder() != nil, item.GetPackage() != nil:
+			var prevPath path.Path
+
+			prevColPath, ok := oldPaths[*item.GetId()]
+			if ok {
+				prevPath, err = path.FromDataLayerPath(prevColPath, false)
+				if err != nil {
+					logger.Ctx(ctx).Errorw("invalid path for deleted item", "error", err)
+					return err
+				}
+			}
+
 			if item.GetDeleted() != nil {
+				if !ok {
```

Comment:
nit: can you have conditional branches based on `prevPath` instead? It feels a little more descriptive than `ok` when reading the code 😅 
```suggestion
				if prevPath == nil {
```